tests/Base.php

```diff
@@ -207,14 +212,40 @@ public function testRequestMethod(): void
 
         $response = $this->execute(method: 'OPTIONS', headers: ['x-action' => 'requestMethod']);
         self::assertEquals(200, $response['code']);
-        // Bug in C++ framework makes this an empty string
-        // self::assertEquals('OPTIONS', $response['body']);
+
+        if($this->runtime === 'cpp') {
+            // Bug in C++ framework
+            self::assertEquals('', $response['body']);
+        } else {
+            self::assertEquals('OPTIONS', $response['body']);
+        }
 
         $response = $this->execute(method: 'PATCH', headers: ['x-action' => 'requestMethod']);
         self::assertEquals(200, $response['code']);
         self::assertEquals('PATCH', $response['body']);
     }
 
+    public function testHeadAndTraceRequestMethods(): void
+    {
+        $response = $this->execute(method: 'HEAD', headers: ['x-action' => 'requestHeadOrTrace']);
+        self::assertEquals(200, $response['code']);
+
+        // head, trace dont work well with cpp [drogon].
+        if($this->runtime === 'cpp') return;
```

Comment:
Not sure if I like this, I would rather have 2 larget if/else, so I know what exactly is expected for CPP. Here it seems to be expected to give 200, but not proper logs.

Reply:
Only the response-code works for CPP, so for that runtime, I returned instead of moving ahead with other tests as they'd eventually be skipped due to an `if/else` check. But I'll add anyway. Better if we have descriptive logs/comments.

Reply:
Understandable, I would prefer this structrure of the test then:

```
if runtiem is cpp
	call()
	ensure 200()
else
	call()
	ensure 200()
	more tests...
```

Its a bit of code duplication but it's for edge-case that isn't supposed to be there, so it's good it stands out.

Reply:
Agree. Will address this too.

Reply:
Done.